### PR TITLE
feat: tricrypto migration [WEB-687]

### PR DESCRIPTION
**Pull request by xgambitox**
- Add crvTriCrypto vault migration.

### CODE REVIEW

**Comment by yearn-integrations** [comment]
## Important changes detected  (payment related code)
- Found constant `TRUSTED_MIGRATOR_ADDRESS` in files app/components/MigrateVault/index.js (`-import { TRUSTED_MIGRATOR_ADDRESS } from 'containers/Vaults/constants';`), app/components/MigrateVault/index.js (`+  TRUSTED_MIGRATOR_ADDRESS,`)  
- Found constant `TRICRYPTO_VAULT_MIGRATOR` in files app/components/MigrateVault/index.js (`+  TRICRYPTO_VAULT_MIGRATOR,`), app/components/MigrateVault/index.js (`+  const triCryptoVaultMigratorContract = useContract(TRICRYPTO_VAULT_MIGRATOR);`), app/containers/App/saga.js (`+  TRICRYPTO_VAULT_MIGRATOR,`), app/containers/App/saga.js (`+    addresses: [TRICRYPTO_VAULT_MIGRATOR],`), app/containers/App/saga.js (`+    TRICRYPTO_VAULT_MIGRATOR,`), app/containers/Vaults/constants.js (`+export const TRICRYPTO_VAULT_MIGRATOR =`), app/containers/Vaults/saga.js (`+  TRICRYPTO_VAULT_MIGRATOR,`), app/containers/Vaults/saga.js (`+    selectTokenAllowance(vaultContract.address, TRICRYPTO_VAULT_MIGRATOR),`)  
- Found constant `TRICRYPTO_VAULT` in files app/components/MigrateVault/index.js (`+  TRICRYPTO_VAULT,`), app/components/MigrateVault/index.js (`+  const isVaultTriCrypto = vaultAddress === TRICRYPTO_VAULT;`), app/components/Vault/index.js (`-  if (vault.address === TRICRYPTO_VAULT) {`), app/components/Vault/index.js (`+    vault.address === TRICRYPTO_VAULT ||`), app/containers/App/saga.js (`+  TRICRYPTO_VAULT,`), app/containers/App/saga.js (`+    [TRICRYPTO_VAULT],`), app/containers/Vaults/selectors.js (`+import { TRICRYPTO_VAULT } from './constants';`), app/containers/Vaults/selectors.js (`+      (migration) => migration.vaultFrom === TRICRYPTO_VAULT,`)  
- Found constant `MIGRATE_TRYCRIPTO_VAULT` in files app/containers/Vaults/actions.js (`+    type: c.MIGRATE_TRYCRIPTO_VAULT,`), app/containers/Vaults/constants.js (`+export const MIGRATE_TRYCRIPTO_VAULT = 'MIGRATE_TRICRYPTO_VAULT';`), app/containers/Vaults/saga.js (`+  MIGRATE_TRYCRIPTO_VAULT,`), app/containers/Vaults/saga.js (`+  yield takeLatest(MIGRATE_TRYCRIPTO_VAULT, migrateTriCryptoVault);`)  
- Found constant `MIGRATE_TRICRYPTO_VAULT` in files app/containers/Vaults/constants.js (`+export const MIGRATE_TRYCRIPTO_VAULT = 'MIGRATE_TRICRYPTO_VAULT';`)  
- Found keyword `cacheSend` in files app/containers/Vaults/saga.js (`+      triCryptoVaultMigratorContract.methods.migrate_to_new_vault.cacheSend,`)  
- Found address [0xE537B5cc158EB71037D4125BDD7538421981E6AA](https://etherscan.io/address/0xE537B5cc158EB71037D4125BDD7538421981E6AA) in files app/containers/Vaults/constants.js, app/containers/Vaults/customOrder.json, app/containers/Vaults/migrationWhitelist.json. This address is from vault `yvCurve-3Crypto`  
- Found address [0xC306a5ef4B990A7F2b3bC2680E022E6a84D75fC1](https://etherscan.io/address/0xC306a5ef4B990A7F2b3bC2680E022E6a84D75fC1) in files app/containers/Vaults/constants.js. This address was ⚠️not found⚠️ in any vaults from https://vaults.finance/all  
- Found address [0xcA3d75aC011BF5aD07a98d02f18225F9bD9A6BDF](https://etherscan.io/address/0xcA3d75aC011BF5aD07a98d02f18225F9bD9A6BDF) in files app/containers/Vaults/migrationWhitelist.json. This address is from token `crvTricrypto`  
- Found address [0x3D980E50508CFd41a13837A60149927a11c03731](https://etherscan.io/address/0x3D980E50508CFd41a13837A60149927a11c03731) in files app/containers/Vaults/migrationWhitelist.json. This address is from vault `yvCurve-triCrypto`

**Comment by yearn-integrations** [comment]
## Important changes detected  (payment related code)
- Found constant `TRUSTED_MIGRATOR_ADDRESS` in files app/components/MigrateVault/index.js (`-import { TRUSTED_MIGRATOR_ADDRESS } from 'containers/Vaults/constants';`), app/components/MigrateVault/index.js (`+  TRUSTED_MIGRATOR_ADDRESS,`)  
- Found constant `TRICRYPTO_VAULT_MIGRATOR` in files app/components/MigrateVault/index.js (`+  TRICRYPTO_VAULT_MIGRATOR,`), app/components/MigrateVault/index.js (`+  const triCryptoVaultMigratorContract = useContract(TRICRYPTO_VAULT_MIGRATOR);`), app/containers/App/saga.js (`+  TRICRYPTO_VAULT_MIGRATOR,`), app/containers/App/saga.js (`+    addresses: [TRICRYPTO_VAULT_MIGRATOR],`), app/containers/App/saga.js (`+    TRICRYPTO_VAULT_MIGRATOR,`), app/containers/Vaults/constants.js (`+export const TRICRYPTO_VAULT_MIGRATOR =`), app/containers/Vaults/saga.js (`+  TRICRYPTO_VAULT_MIGRATOR,`), app/containers/Vaults/saga.js (`+    selectTokenAllowance(vaultContract.address, TRICRYPTO_VAULT_MIGRATOR),`)  
- Found constant `TRICRYPTO_VAULT` in files app/components/MigrateVault/index.js (`+  TRICRYPTO_VAULT,`), app/components/MigrateVault/index.js (`+  const isVaultTriCrypto = vaultAddress === TRICRYPTO_VAULT;`), app/components/Vault/index.js (`-  if (vault.address === TRICRYPTO_VAULT) {`), app/components/Vault/index.js (`+    vault.address === TRICRYPTO_VAULT ||`), app/containers/App/saga.js (`+  TRICRYPTO_VAULT,`), app/containers/App/saga.js (`+    [TRICRYPTO_VAULT],`), app/containers/Vaults/selectors.js (`+import { TRICRYPTO_VAULT } from './constants';`), app/containers/Vaults/selectors.js (`+      (migration) => migration.vaultFrom === TRICRYPTO_VAULT,`)  
- Found constant `MIGRATE_TRYCRIPTO_VAULT` in files app/containers/Vaults/actions.js (`+    type: c.MIGRATE_TRYCRIPTO_VAULT,`), app/containers/Vaults/constants.js (`+export const MIGRATE_TRYCRIPTO_VAULT = 'MIGRATE_TRICRYPTO_VAULT';`), app/containers/Vaults/saga.js (`+  MIGRATE_TRYCRIPTO_VAULT,`), app/containers/Vaults/saga.js (`+  yield takeLatest(MIGRATE_TRYCRIPTO_VAULT, migrateTriCryptoVault);`)  
- Found constant `MIGRATE_TRICRYPTO_VAULT` in files app/containers/Vaults/constants.js (`+export const MIGRATE_TRYCRIPTO_VAULT = 'MIGRATE_TRICRYPTO_VAULT';`)  
- Found keyword `cacheSend` in files app/containers/Vaults/saga.js (`+      triCryptoVaultMigratorContract.methods.migrate_to_new_vault.cacheSend,`)  
- Found address [0xE537B5cc158EB71037D4125BDD7538421981E6AA](https://etherscan.io/address/0xE537B5cc158EB71037D4125BDD7538421981E6AA) in files app/containers/Vaults/constants.js, app/containers/Vaults/customOrder.json, app/containers/Vaults/migrationWhitelist.json. This address is from vault `yvCurve-3Crypto`  
- Found address [0xC306a5ef4B990A7F2b3bC2680E022E6a84D75fC1](https://etherscan.io/address/0xC306a5ef4B990A7F2b3bC2680E022E6a84D75fC1) in files app/containers/Vaults/constants.js. This address was ⚠️not found⚠️ in any vaults from https://vaults.finance/all  
- Found address [0xcA3d75aC011BF5aD07a98d02f18225F9bD9A6BDF](https://etherscan.io/address/0xcA3d75aC011BF5aD07a98d02f18225F9bD9A6BDF) in files app/containers/Vaults/migrationWhitelist.json. This address is from token `crvTricrypto`  
- Found address [0x3D980E50508CFd41a13837A60149927a11c03731](https://etherscan.io/address/0x3D980E50508CFd41a13837A60149927a11c03731) in files app/containers/Vaults/migrationWhitelist.json. This address is from vault `yvCurve-triCrypto`

File: app/containers/Vaults/migrationWhitelist.json [hunk 1]
```diff
@@ -238,5 +238,13 @@
     "vaultTo": "0xdA816459F1AB5631232FE5e97a05BBBb94970c95",
     "apyTooltip": "Please migrate funds to our new vault to continue earning yield.",
     "migrationMessage": "This vault is winding down. Withdraw funds or migrate your balance to the new v2 DAI yVault to continue earning yield."
+  },
+  {
+    "symbol": "crvTriCrypto",
+    "token": "0xcA3d75aC011BF5aD07a98d02f18225F9bD9A6BDF",
+    "vaultFrom": "0x3D980E50508CFd41a13837A60149927a11c03731",
+    "vaultTo": "0xE537B5cc158EB71037D4125BDD7538421981E6AA",
```
Comment (x48-crypto) [inline app/containers/Vaults/migrationWhitelist.json:246]: @dudesahn can you confirm these are correct? I looked and it looks good, just want to double check

**Review by x48-crypto — APPROVED** [review]

**Comment by dudesahn** [comment]
@xgambitox could we remove the blue box from the old crvTricrypto vault here:

Screen Shot 2021-08-13 at 4 16 10 PM

**Comment by dudesahn** [comment]
The zap/migration from the UI works great, though! Awesome work, if you could just remove the extra blue warning box on the old vault and we should we good to merge :)

**Comment by yearn-integrations** [comment]
## Important changes detected  (payment related code)
- Found constant `TRUSTED_MIGRATOR_ADDRESS` in files app/components/MigrateVault/index.js (`-import { TRUSTED_MIGRATOR_ADDRESS } from 'containers/Vaults/constants';`), app/components/MigrateVault/index.js (`+  TRUSTED_MIGRATOR_ADDRESS,`)  
- Found constant `TRICRYPTO_VAULT_MIGRATOR` in files app/components/MigrateVault/index.js (`+  TRICRYPTO_VAULT_MIGRATOR,`), app/components/MigrateVault/index.js (`+  const triCryptoVaultMigratorContract = useContract(TRICRYPTO_VAULT_MIGRATOR);`), app/containers/App/saga.js (`+  TRICRYPTO_VAULT_MIGRATOR,`), app/containers/App/saga.js (`+    addresses: [TRICRYPTO_VAULT_MIGRATOR],`), app/containers/App/saga.js (`+    TRICRYPTO_VAULT_MIGRATOR,`), app/containers/Vaults/constants.js (`+export const TRICRYPTO_VAULT_MIGRATOR =`), app/containers/Vaults/saga.js (`+  TRICRYPTO_VAULT_MIGRATOR,`), app/containers/Vaults/saga.js (`+    selectTokenAllowance(vaultContract.address, TRICRYPTO_VAULT_MIGRATOR),`)  
- Found constant `TRICRYPTO_VAULT` in files app/components/MigrateVault/index.js (`+  TRICRYPTO_VAULT,`), app/components/MigrateVault/index.js (`+  const isVaultTriCrypto = vaultAddress === TRICRYPTO_VAULT;`), app/components/Vault/index.js (`-  TRICRYPTO_VAULT,`), app/components/Vault/index.js (`-  if (vault.address === TRICRYPTO_VAULT) {`), app/containers/App/saga.js (`+  TRICRYPTO_VAULT,`), app/containers/App/saga.js (`+    [TRICRYPTO_VAULT],`), app/containers/Vaults/selectors.js (`+import { TRICRYPTO_VAULT } from './constants';`), app/containers/Vaults/selectors.js (`+      (migration) => migration.vaultFrom === TRICRYPTO_VAULT,`)  
- Found constant `MIGRATE_TRYCRIPTO_VAULT` in files app/containers/Vaults/actions.js (`+    type: c.MIGRATE_TRYCRIPTO_VAULT,`), app/containers/Vaults/constants.js (`+export const MIGRATE_TRYCRIPTO_VAULT = 'MIGRATE_TRICRYPTO_VAULT';`), app/containers/Vaults/saga.js (`+  MIGRATE_TRYCRIPTO_VAULT,`), app/containers/Vaults/saga.js (`+  yield takeLatest(MIGRATE_TRYCRIPTO_VAULT, migrateTriCryptoVault);`)  
- Found constant `MIGRATE_TRICRYPTO_VAULT` in files app/containers/Vaults/constants.js (`+export const MIGRATE_TRYCRIPTO_VAULT = 'MIGRATE_TRICRYPTO_VAULT';`)  
- Found keyword `cacheSend` in files app/containers/Vaults/saga.js (`+      triCryptoVaultMigratorContract.methods.migrate_to_new_vault.cacheSend,`)  
- Found address [0xE537B5cc158EB71037D4125BDD7538421981E6AA](https://etherscan.io/address/0xE537B5cc158EB71037D4125BDD7538421981E6AA) in files app/containers/Vaults/constants.js, app/containers/Vaults/customOrder.json, app/containers/Vaults/migrationWhitelist.json. This address is from vault `yvCurve-3Crypto`  
- Found address [0xC306a5ef4B990A7F2b3bC2680E022E6a84D75fC1](https://etherscan.io/address/0xC306a5ef4B990A7F2b3bC2680E022E6a84D75fC1) in files app/containers/Vaults/constants.js. This address was ⚠️not found⚠️ in any vaults from https://vaults.finance/all  
- Found address [0xcA3d75aC011BF5aD07a98d02f18225F9bD9A6BDF](https://etherscan.io/address/0xcA3d75aC011BF5aD07a98d02f18225F9bD9A6BDF) in files app/containers/Vaults/migrationWhitelist.json. This address is from token `crvTricrypto`  
- Found address [0x3D980E50508CFd41a13837A60149927a11c03731](https://etherscan.io/address/0x3D980E50508CFd41a13837A60149927a11c03731) in files app/containers/Vaults/migrationWhitelist.json. This address is from vault `yvCurve-triCrypto`

**Comment by xgambitox** [comment]
Done. Removed the blue box just for the old crvTricrypto vault.

**Review by dudesahn — APPROVED** [review]

**Pull request merged**